adapters/ddf-adapter/src/main/java/com/connexta/replication/adapters/ddf/DdfNodeAdapter.java

```diff
@@ -0,0 +1,401 @@
+      return false;
+    }
+    return true;
+  }
+
+  private List<Metadata> getRecords(QueryRequest request) {
+    try {
+      Csw csw = factory.getClient();
+
+      GetRecordsType getRecordsType = new GetRecordsType();
+      getRecordsType.setVersion(Constants.VERSION_2_0_2);
+      getRecordsType.setService("CSW");
+      getRecordsType.setResultType(ResultType.RESULTS);
+      getRecordsType.setStartPosition(BigInteger.valueOf(request.getStartIndex()));
+      getRecordsType.setMaxRecords(BigInteger.valueOf(request.getPageSize()));
+      getRecordsType.setOutputFormat(MediaType.APPLICATION_XML);
+      getRecordsType.setOutputSchema("urn:catalog:metacard");
+      QueryType queryType = new QueryType();
+      queryType.setTypeNames(
+          Arrays.asList(new QName("http://www.opengis.net/cat/csw/2.0.2", "Record", "csw")));
+
+      ElementSetNameType elementSetNameType = new ElementSetNameType();
+      elementSetNameType.setValue(ElementSetType.FULL);
+      queryType.setElementSetName(elementSetNameType);
+      SortByType cswSortBy = new SortByType();
+      SortPropertyType sortProperty = new SortPropertyType();
+      PropertyNameType propertyName = new PropertyNameType();
+      propertyName.setContent(Arrays.asList(METACARD_MODIFIED));
+      sortProperty.setPropertyName(propertyName);
+      sortProperty.setSortOrder(SortOrderType.ASC);
+      cswSortBy.getSortProperty().add(sortProperty);
+      queryType.setSortBy(cswSortBy);
+      QueryConstraintType queryConstraintType = new QueryConstraintType();
+      queryConstraintType.setVersion(Constants.CONSTRAINT_VERSION);
+      queryConstraintType.setCqlText(request.getCql());
+      queryType.setConstraint(queryConstraintType);
+      ObjectFactory objectFactory = new ObjectFactory();
+      getRecordsType.setAbstractQuery(objectFactory.createQuery(queryType));
+
+      CswRecordCollection response = csw.getRecords(getRecordsType);
+      return response.getCswRecords();
+    } catch (Exception e) {
+      throw new AdapterException("Error executing csw getRecords", e);
+    }
+  }
+
+  @Override
+  public String getSystemName() {
+    if (this.cachedSystemName != null) {
+      return this.cachedSystemName;
+    }
+    String filter =
+        String.format(
+            "[[ \"%s\" = '%s' ] AND [ NOT [ \"%s\" IS NULL ] ]]",
+            METACARD_TAGS, REGISTRY_TAG, REGISTRY_IDENTITY_NODE);
+    List<Metadata> results;
+
+    try {
+      results = getRecords(new QueryRequestImpl(filter, 1, 1));
+    } catch (Exception e) {
+      throw new AdapterException("Failed to retrieve remote system name", e);
+    }
+
+    String systemName;
+    if (!results.isEmpty()) {
+      systemName =
+          ((MetacardAttribute) ((Map) results.get(0).getRawMetadata()).get("title")).getValue();
+    } else {
+      throw new AdapterException(
+          "No registry metadata available on remote node. Could not retrieve remote system name");
```

Comment:
Should we add the hostUrl to some of these exception messages to clarify which source is having the issue?